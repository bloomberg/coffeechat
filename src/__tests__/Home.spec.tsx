import { render } from '@testing-library/react'
import { Session } from 'next-auth'
import { SessionProvider, useSession } from 'next-auth/react'
import { FC } from 'react'
import Home from '../pages/home'

jest.mock('next/router', () => ({
  push: jest.fn(),
  useRouter: jest.fn(),
}))

jest.mock('next-auth/react', () => ({
  useSession: jest.fn().mockReturnValue({ data: null }),
  SessionProvider: jest
    .fn()
    .mockImplementation(({ children }) => <>{children}</>),
  signOut: jest.fn(),
}))

describe('Home Page', () => {
  let container: HTMLElement

  const WrappedHome: FC = () => {
    return (
      <SessionProvider>
        <Home />
      </SessionProvider>
    )
  }
  describe('session exists', () => {
    beforeEach(async () => {
      const data: Session = {
        user: {
          name: 'Test Testoff',
        },
      } as Session
      ;(useSession as jest.Mock).mockReturnValue({ data })
      ;({ container } = render(<WrappedHome />))
    })

    it('displays username', () => {
      expect(container).toMatchSnapshot()
    })
    it('calls useSession with required Args', () => {
      expect(useSession).toHaveBeenCalledWith({
        required: true,
        onUnauthenticated: expect.any(Function) as () => void,
      })
    })
  })
})
