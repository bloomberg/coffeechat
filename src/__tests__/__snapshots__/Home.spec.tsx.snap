// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page session exists displays username 1`] = `
<div>
  <div
    class="p-10"
  >
    <h2>
      Welcome to the homepage
    </h2>
    <p>
      Test Testoff
    </p>
  </div>
</div>
`;
